Core/Src/currentSense.c

```diff
@@ -0,0 +1,59 @@
+/* ==================================================================== */
+/* ============================= INCLUDES ============================= */
+/* ==================================================================== */
+
+#include "currentSense.h"
+#include "GopherCAN.h"
+#include <math.h>
+#include "debug.h"
+
+/* ==================================================================== */
+/* =================== GLOBAL FUNCTION DEFINITIONS ==================== */
+/* ==================================================================== */
+
+/*!
+  @brief   Update the bms tractive system current data and sensor status from gophercan
+  @param   bms - BMS data struct
+*/
+void getTractiveSystemCurrent(Bms_S* bms)
+{
+    // Fetch current sensor data from gophercan
+    float currHI = bmsTractiveSystemCurrentHigh_A.data;
+    float currLO = bmsTractiveSystemCurrentLow_A.data;
+
+    // If the current exceeds the following threshold in either the positive or negative direction,
+    // the sensor input has railed to 0 or 5v and a current sensor error is set
+    if(fabs(currHI) < CURRENT_HIGH_RAIL_THRESHOLD)
```

Comment:
Might be simpler to just replace with ternary operator
`bms->currentSensorStatusHI = (fabs(currHI) < CURRENT_HIGH_RAIL_THRESHOLD) ? SENSE_GOOD : SENSE_MIA;`